src/api/tigl.cpp

```diff
@@ -6501,6 +6507,98 @@ TIGL_COMMON_EXPORT TiglReturnCode tiglFuselageGetSegmentVolume(TiglCPACSConfigur
 /*                     Surface Area calculations                                                     */
 /*****************************************************************************************************/
 
+
+TIGL_COMMON_EXPORT TiglReturnCode tiglGetCrossSectionArea(TiglCPACSConfigurationHandle cpacsHandle,
+                                                          const char* componentUID,
+                                                          double origin_x, double origin_y, double origin_z,
+                                                          double normal_x, double normal_y, double normal_z,
+                                                          double* area
+                                                          )
+{
+
+    try {
+
+        //get configuration and uID manager
+
+        tigl::CCPACSConfigurationManager& manager = tigl::CCPACSConfigurationManager::GetInstance();
+        tigl::CCPACSConfiguration& config = manager.GetConfiguration(cpacsHandle);
+        tigl::CTiglUIDManager& uIDManager = config.GetUIDManager();
+
+        if (normal_x*normal_x + normal_y*normal_y + normal_z*normal_z < 1e-15) {
+            LOG(ERROR) << "Plane normal must be non-zero.";
+            return TIGL_MATH_ERROR;
+        }
+        if (uIDManager.HasGeometricComponent(componentUID) == false) {
+            LOG(ERROR) << "uID doesn't refer to a geometric component.";
+            return TIGL_UID_ERROR;
+        }
+
+        *area = 0.;
+
+        // create plane
+
+        gp_Pnt point = gp_Pnt(origin_x, origin_y, origin_z);
+        gp_Dir normal = gp_Dir(normal_x, normal_y, normal_z);
+
+        TopoDS_Shape planeSurface = BRepBuilderAPI_MakeFace(gp_Pln(point, normal));
+
+        if (componentUID == config.GetUID()) {
+
+            // get the fused airplane
+
+            tigl::PTiglFusePlane fuser = config.AircraftFusingAlgo();
+            fuser->SetResultMode((tigl::TiglFuseResultMode) 1);
+            PNamedShape airplane = fuser->FusedPlane();
+            TopoDS_Shape airplaneShape = airplane->Shape();
+
+            // compute intersection of fused airplane and cutting plane
+
+            TopoDS_Shape commonSurface = BRepAlgoAPI_Common(planeSurface, airplaneShape);
+
+            // calculate intersection area
+
+            GProp_GProps props = GProp_GProps();
+            BRepGProp::SurfaceProperties(commonSurface, props);
+            double val = props.Mass();
+            *area =val;
+            return TIGL_SUCCESS;
```

Comment:
This code is duplicated in the `else` block. Can We move this to *after* the `else`-block to avoid code duplication?

Reply:
Oh yes, sure